fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveConnectorScanRangeSource.java

```diff
@@ -0,0 +1,409 @@
+                op = scalarRewriter.rewrite(op, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
+                if (op.isConstantTrue()) {
+                    // matched partition predicates
+                    dataCacheOptions.add(DataCacheOptions.DataCacheOptionsBuilder.builder()
+                            .setPriority(dataCacheRule.get().getPriority()).build());
+                } else {
+                    // not matched, add null DataCacheOption
+                    dataCacheOptions.add(null);
+                    if (!op.isConstantRef()) {
+                        LOG.warn(String.format("ConstFolding failed for expr: %s, rewrite scalarOperator is %s",
+                                rewritedExpr.toMySql(), op.debugString()));
+                    }
+                }
+            }
+        }
+        return Optional.of(dataCacheOptions);
+    }
+
+    class ScanRangeIterator implements Iterator<TScanRangeLocations> {
+        RemoteFileInfo remoteFileInfo;
+        List<RemoteFileDesc> files;
+        int fileIndex = 0;
+        List<TScanRangeLocations> buffer = new ArrayList<>();
+
+        public ScanRangeIterator(RemoteFileInfo remoteFileInfo) {
+            this.remoteFileInfo = remoteFileInfo;
+            this.files = remoteFileInfo.getFiles();
+        }
+
+        @Override
+        public boolean hasNext() {
+            tryPopulateBuffer();
+            return fileIndex < files.size() || !buffer.isEmpty();
+        }
+
+        @Override
+        public TScanRangeLocations next() {
+            return buffer.remove(buffer.size() - 1);
+        }
+
+        private void tryPopulateBuffer() {
+            while (buffer.isEmpty() && fileIndex < files.size()) {
+                splitFile(files.get(fileIndex));
+                fileIndex += 1;
+            }
+        }
+
+        private void splitFile(RemoteFileDesc fileDesc) {
+            if (forceScheduleLocal) {
+                for (RemoteFileBlockDesc blockDesc : fileDesc.getBlockDescs()) {
+                    addScanRangeLocations(fileDesc, Optional.of(blockDesc));
+                    LOG.debug("Add scan range success. partition: {}, file: {}, block: {}-{}",
+                            remoteFileInfo.getFullPath(), fileDesc.getFileName(), blockDesc.getOffset(),
+                            blockDesc.getLength());
+                }
+            } else {
+                addScanRangeLocations(fileDesc, Optional.empty());
+                LOG.debug("Add scan range success. partition: {}, file: {}, range: {}-{}",
+                        remoteFileInfo.getFullPath(), fileDesc.getFileName(), 0, fileDesc.getLength());
+            }
+        }
+
+        private void addScanRangeLocations(RemoteFileDesc fileDesc, Optional<RemoteFileBlockDesc> blockDesc) {
+            SessionVariable sv = SessionVariable.DEFAULT_SESSION_VARIABLE;
+            ConnectContext connectContext = ConnectContext.get();
+            if (connectContext != null) {
```

Comment:
SessionVariable sv = connectContext != null? connectContext : SessionVariable.DEFAULT_SESSION_VARIABLE;

Reply:
do you think it's more concise?

Reply:
I think it will only new session variable when necessary

Reply:
> public static final SessionVariable DEFAULT_SESSION_VARIABLE = new SessionVariable();

DEFAULT_SESSION_VARIABLE is a static member, it's created anyway.